applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp

```diff
@@ -1039,6 +1164,46 @@ void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::SelectUpwindEle
     }
 }
 
+template<int TDim, int TNumNodes>
+int TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::GetAdditionalNode() const
+{
+    const TransonicPerturbationPotentialFlowElement& r_this = *this;
+
+    // current and upwind element geometry
+    const GeometryType& r_geom = r_this.GetGeometry();
+    const GeometryType& r_upwind_geom = mpUpwindElement->GetGeometry();
```

Comment:
It is safer to use pGetUpwindElement
```suggestion
    const GeometryType& r_upwind_geom = pGetUpwindElement()->GetGeometry();
```